lib/model-builder.js

```diff
@@ -369,28 +409,81 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     }
 
     // Merging the properties
-    var keys = Object.keys(properties);
+    var keys = Object.keys(baseclassProperties);
     for (var i = 0, n = keys.length; i < n; i++) {
       var key = keys[i];
 
-      if (idFound && properties[key].id) {
+      if (idFound && baseclassProperties[key].id) {
         // don't inherit id properties
         continue;
       }
       if (subclassProperties[key] === undefined) {
-        var baseProp = properties[key];
+        var baseProp = baseclassProperties[key];
         var basePropCopy = baseProp;
         if (baseProp && typeof baseProp === 'object') {
-          // Deep clone the base prop
-          basePropCopy = mergeSettings(null, baseProp);
```

Comment:
`mergeSettings(baseProp)` is fully equivalent to` mergeSettings(null, baseProp)` and operates a deep clone.
`mergeSettings(baseProp)` is more readable